### ISSUE TITLE
SaveFig plot naming scheme and auto_increment

### ISSUE THREAD

**Issue by Areustle** [NuSpaceSim/nuSpaceSim#20]
When plots are generated they always save the file as the default matplotlib savefig name "Figure_1.png" or something like that.

Proposal:

- [ ] Provide a way to customize name prefix for saves plot images.
- [ ] Set a default name. Probably default to the simulation output file name like `nuspacesim_run_<timestamp>_<number>.png`
- [ ] Auto increment <number>: 1, 2, ...

**Comment by Areustle** [comment]
Attention of @jfkrizmanic and @smayotte, any thoughts?

**Comment by smayotte** [comment]
I think this is really important. On the default filename maybe it would make more sense to have the default name of a plot be the name of the respective plotting function? `nuspacesim_run_<timestamp>_<number>.png` would make it a bit difficult to quickly distinguish plots. If we go with something like `<plot_function_name>_<timestamp>.png` that would make it a lot clearer (albeit maybe not as straightforward in implementation)

**Comment by Areustle** [comment]
I think that's a good point @smayotte . How about a scheme like `<simulation_name>_<plot_function_name>.png` where simulation_name is the same as the user defined output FITS file name with its existing default value.

**Comment by smayotte** [comment]
I spent some time working on this and have a working implementations for both the `--plotconfig` and `--plot` options. I hope the implementation is not too clunky. Should I commit that to a branch again so that you can review it?

**Comment by Areustle** [comment]
Yes you should commit it to a branch and open a pull request. Reference this issue by including the string "#20" in the PR text and we'll start a code review.
